docs/docs/handlers-and-http/generic-handlers.md

```diff
@@ -138,4 +138,88 @@ end
 
 By default, such a handler will render the configured template when the incoming request is a GET or for POST requests if the data cannot be validated using the specified schema (in that case, the template is expected to use the invalid schema to display a form with the right errored inputs). The specified template can have access to the configured schema through the use of the `schema` object in the template context.
 
-If the schema is valid, a temporary redirect is issued by using the URL corresponding to the `#success_route_name` value (although it should be noted that the way to generate this success URL can be overridden by defining a `#success_url` method). By default, the handler does nothing when the processed schema is valid (except redirecting to the success URL). That's why it can be helpful to override the `#process_valid_schema` method to implement any logic that should be triggered after a successful schema validation.
+If the schema is valid, a temporary redirect is issued by using the URL corresponding to the `#success_route_name` value (although it should be noted that the way to generate this success URL can be overridden by defining a `#success_url` method). By default, the handler does nothing when the processed schema is valid (except redirecting to the success URL).
+
+
+#### Callbacks
+
+Callbacks let you define logics that are triggered before or after the validation of the schema. This allows you to easily intercept validation
+and handle the response independently of the schema validty. Two callbacks are supported: `before_schema_validation`, `after_schema_validation`,
+`after_successful_schema_validation` and `after_failed_schema_validation`.
+
+All callbacks can optionally return a [`Marten::HTTP::Response`](pathname:///api/dev/Marten/HTTP/Response.html). When a HTTP response is returned
```

Comment:
```suggestion
All callbacks can optionally return a [`Marten::HTTP::Response`](pathname:///api/dev/Marten/HTTP/Response.html). When an HTTP response is returned,
```

Reply:
Is it really **an response**? I'm pretty sure its **a response**